tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -127,7 +129,14 @@ public void onSuccess(RawResponse response) {
             @Override
             public void onFailure(Throwable throwable) {
                 queuedRequests.decrementAndGet();
-                // TODO handle the failure case
+
+                // TODO better interface for sending errors
+                ErrorMessage error = new ErrorMessage(
+                        rawRequest.getId(),
+                        ErrorType.BadRequest,
```

Comment:
not sure which function you refer to, but, yes, this is the right response. Essentially, it cannot decode arg2 or arg3, it is quite a BadRequest.